js/import/import.ui.js

```diff
@@ -427,6 +427,35 @@ const createImporter = () => {
 
 const getContentFrame = () => document.querySelector(`${PARENT_SELECTOR} iframe`);
 
+/**
+ * After an import or detect operation, return the UI to the waiting state.  Ensure all listeners
+ * are removed and the buttons are re-enabled.
+ * @param processNext: function to remove 'transformation-complete' listener from
+ * @param finishingImport: whether an import is finishing (or detect)
+ */
+const restoreWaitingUI = (processNext, finishingImport) => {
+  if (processNext) {
+    getContentFrame().removeEventListener('transformation-complete', processNext);
+  }
+
+  DOWNLOAD_IMPORT_REPORT_BUTTON?.classList.remove('hidden');
+  DOWNLOAD_TRANSFORMATION_BUTTON?.classList.remove('hidden');
```

Comment:
Maybe these ☝️ should only be shown after an 'import'?  If they do a 'detect' again, are the 2 reports still valid?
Maybe more complicated would be to invalidate them if they change the mapping and/or customizations. 🤷

Reply:
Good question. If the mapping object exists then a transformation ruleset can be created from it. Do we always allow download or are there certain cases where even though the mpaping objects exists we should consider it invalid?